segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/ContainerRecoveryUtils.java

```diff
@@ -418,33 +419,32 @@ public static void updateCoreAttributes(Map<Integer, String> backUpMetadataSegme
-            throws InterruptedException, ExecutionException, TimeoutException {
+    public static synchronized CompletableFuture<Map<Integer, String>> createBackUpMetadataSegments(Storage storage, int containerCount, ExecutorService executorService,
+                                                                    Duration timeout) {
         String fileSuffix = new SimpleDateFormat("yyyyMMddHHmmss").format(new Date());
         Map<Integer, String> backUpMetadataSegments = new HashMap<>();
 
         val futures = new ArrayList<CompletableFuture<Void>>();
 
-        for (int containerId = 0; containerId < containerCount; containerId++) {
-            String backUpMetadataSegment = NameUtils.getMetadataSegmentName(containerId) + fileSuffix;
-            String backUpAttributeSegment = NameUtils.getAttributeSegmentName(backUpMetadataSegment);
-            log.debug("Created '{}' as a back of metadata segment of container Id '{}'", backUpMetadataSegment, containerId);
-
-            val finalContainerId = containerId;
-            futures.add(Futures.exceptionallyExpecting(
-                    ContainerRecoveryUtils.backUpMetadataAndAttributeSegments(storage, containerId,
-                            backUpMetadataSegment, backUpAttributeSegment, executorService, timeout)
-                            .thenAccept(x -> ContainerRecoveryUtils.deleteMetadataAndAttributeSegments(storage, finalContainerId, timeout)
-                                    .thenAccept(z -> backUpMetadataSegments.put(finalContainerId, backUpMetadataSegment))
-                            ), ex -> Exceptions.unwrap(ex) instanceof StreamSegmentNotExistsException, null));
-        }
-        Futures.allOf(futures).get(timeout.toMillis(), TimeUnit.MILLISECONDS);
-        return backUpMetadataSegments;
+        return CompletableFuture.supplyAsync(() -> {
+            for (int containerId = 0; containerId < containerCount; containerId++) {
+                String backUpMetadataSegment = NameUtils.getMetadataSegmentName(containerId) + fileSuffix;
+                String backUpAttributeSegment = NameUtils.getAttributeSegmentName(backUpMetadataSegment);
+                log.debug("Created '{}' as a back of metadata segment of container Id '{}'", backUpMetadataSegment, containerId);
+
+                val finalContainerId = containerId;
+                futures.add(Futures.exceptionallyExpecting(
+                        ContainerRecoveryUtils.backUpMetadataAndAttributeSegments(storage, containerId,
+                                backUpMetadataSegment, backUpAttributeSegment, executorService, timeout)
+                                .thenCompose(x -> ContainerRecoveryUtils.deleteMetadataAndAttributeSegments(storage, finalContainerId, timeout)),
+                                        ex -> Exceptions.unwrap(ex) instanceof StreamSegmentNotExistsException, null));
+                backUpMetadataSegments.put(finalContainerId, backUpMetadataSegment);
```

Comment:
returning `completableFuture` now. 
`synchroized` to avoid multiple threads from deleting and making copies. 

Issue [#5481(here),](https://github.com/pravega/pravega/issues/5481#issuecomment-800085111) the mismatch between the number of backup metadata segments and containers is also be fixed here. In case, when metadata segment(s) don't exist, it throws `StreamSegmentNotExistsException` and the backup segment created was never put into the map of backup metadata segments. But now, an empty backup metadata segment will be created and put into the map.